libPhoneNumber/NBAsYouTypeFormatter.m

```diff
@@ -810,10 +810,12 @@ - (BOOL)ableToExtractLongerNdd_ {
     NSString *prefixBeforeNationalNumberStr = [self.prefixBeforeNationalNumber_ copy];
     NSRange lastRange = [prefixBeforeNationalNumberStr rangeOfString:self.nationalPrefixExtracted_
                                                              options:NSBackwardsSearch];
-    /** @type {number} */
-    NSUInteger indexOfPreviousNdd = lastRange.location;
-    self.prefixBeforeNationalNumber_ = [[prefixBeforeNationalNumberStr
-        substringWithRange:NSMakeRange(0, indexOfPreviousNdd)] mutableCopy];
+    if (lastRange.length > 0) {
```

Comment:
@constantintin in theory you can have range with a location and a zero length, so normally a valid range is checked by comparing the location to `NSNotFound`, e.g. here `lastRange.location != NSNotFound`

Reply:
I was going off of this comment (`These methods perform string search, looking for the searchString within the receiver string.  These return length==0 if the target string is not found. So, to check for containment: ([str rangeOfString:@"target"].length > 0).  Note that the length of the range returned by these methods might be different than the length of the target string, due composed characters and such.`) found in the h-file for `-[NSString rangeOfString:]`.

Reply:
Should I change this? NSNotFound makes sense to me as well.

Reply:
It shouldn't make a difference in this case.